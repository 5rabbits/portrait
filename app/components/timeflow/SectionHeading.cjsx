React = require('react')

@SectionHeading = React.createClass
  render: ->
    return (
      <div>
        <h1 className="type-title inline">Proyectos</h1>
        <a href="#" className="link type-footnote color-secondary ml-sm">Ver Clientes</a>
      </div>
    )

module.exports = @SectionHeading

