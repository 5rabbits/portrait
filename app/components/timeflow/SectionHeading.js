import React, { PureComponent } from 'react';

export default SectionHeading extends PureComponent {
  render() {
    return (
      <div>
        <h1 className="type-title inline">Proyectos</h1>
        <a href="#" className="link type-footnote color-secondary ml-sm">Ver Clientes</a>
      </div>
    );
  }
}
