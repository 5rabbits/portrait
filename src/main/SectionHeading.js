import React, { PureComponent } from 'react'

export default class SectionHeading extends PureComponent {
  buildLink() {
    if (!this.props.label) {
      return null
    }

    return (
      <a
        href={this.props.url || '#'}
        className="link type-footnote color-secondary ml-sm"
        >
        {this.props.label}
      </a>
    )
  }

  render() {
    return (
      <div>
        <h1 className="type-title inline">{this.props.title}</h1>
        {this.buildLink()}
      </div>
    )
  }
}
