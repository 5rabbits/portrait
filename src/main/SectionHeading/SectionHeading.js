import React, { PureComponent, PropTypes } from 'react'

export default class SectionHeading extends PureComponent {
  static propTypes = {
    label: PropTypes.string,
    url: PropTypes.string,
    title: PropTypes.string,
  }

  buildLink() {
    if (!this.props.label) {
      return null
    }

    return (
      <a
        href={this.props.url || '#'}
        className="link link-border type-footnote color-secondary ml-sm"
        >
        {this.props.label}
      </a>
    )
  }

  render() {
    return (
      <div>
        <h1 className="type-title inline">{this.props.title}</h1>
        {this.buildLink()}
      </div>
    )
  }
}
