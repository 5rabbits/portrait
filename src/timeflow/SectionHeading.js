import React, { PureComponent } from 'react';
import { Link } from 'react-router';

export default class SectionHeading extends PureComponent {
  render() {
    return (
      <div>
        <h1 className="type-title inline">{this.props.title}</h1>
        { this.buildFootnote() }
      </div>
    );
  }

  buildFootnote() {
    if (!this.props.footnote) {
      return null;
    }

    return (
      <Link href={this.props.footnoteUrl || '#'}
         className="link type-footnote color-secondary ml-sm">
        {this.props.footnote}
      </Link>
    );
  }
}
