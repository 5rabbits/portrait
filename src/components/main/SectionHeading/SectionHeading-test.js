import { React, toJSON, mount } from 'test/helper'
import SectionHeading from 'main/SectionHeading'

describe('SectionHeading', () => {
  it('should render with minimum props', () => {
    const component = mount(<SectionHeading />)

    expect(toJSON(component)).toMatchSnapshot()
  })
})
