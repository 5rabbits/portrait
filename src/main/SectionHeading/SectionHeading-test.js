import { React, expect, mount } from 'test/helper'
import SectionHeading from 'main/SectionHeading'

describe('SectionHeading', () => {
  it('should render with minimum props', () => {
    const component = mount(<SectionHeading />)

    expect(component).to.be.ok
  })
})
