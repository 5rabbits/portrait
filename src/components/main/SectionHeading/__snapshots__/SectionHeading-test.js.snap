// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SectionHeading should render with minimum props 1`] = `
<SectionHeading>
  <div>
    <h1
      className="type-title inline"
    />
  </div>
</SectionHeading>
`;
